invoker/core/src/main/java/com/google/cloud/functions/invoker/runner/Invoker.java

```diff
@@ -261,7 +307,21 @@ public void startServer() throws Exception {
 
     server.start();
     logServerInfo();
-    server.join();
+    if (join) {
+      server.join();
+    }
+  }
+
+  /**
+   * Stop the server.
+   *
+   * @see #startServer()
+   * @see #startTestServer()
+   *
+   * @throws Exception
+   */
+  public void stopServer() throws Exception {
+    server.stop();
```

Comment:
And here:
```java
    if (server == null) {
     throw new IllegalStateException("Server not started");
   }
```
(which is a bit nicer than `NullPointerException`).

Reply:
Maybe setting the server to null after stopping it so the same invoker can be stoped and stated again without needed to create it again ?
This can be convenient if someone want to stop it / start it inside a `@Before` and create it inside an `@BeforeAll` ?

Reply:
Yes, thanks for catching that, it should indeed be set to null after the `stop()` call.

Reply:
Changes done